src/evaluators/ClassDeclaration.js

```diff
@@ -10,11 +10,280 @@
+      //     constructor( ){ }
+      // using the syntactic grammar with the goal symbol MethodDefinition.
+      constructorFile = parse(realm, 'class NeedClassForParsing { constructor( ){ } }', '');
+    }
+
+    let { program: { body: [classDeclaration] } } = constructorFile;
+    invariant(classDeclaration.type === "ClassDeclaration");
+    let { body } = ((classDeclaration: any): BabelNodeClassDeclaration);
+    invariant(body.body[0].type === 'ClassMethod');
+    constructor = ((body.body[0]: any): BabelNodeClassMethod);
+  }
+
+  // 11. Set the running execution context’s LexicalEnvironment to classScope.
+  realm.getRunningContext().lexicalEnvironment = classScope;
+
+  let F;
+  try {
+    // 12. Let constructorInfo be the result of performing DefineMethod for constructor with arguments proto and constructorParent as the optional functionPrototype argument.
+    let constructorInfo = DefineMethod(realm, constructor, proto, env, strictCode, constructorParent);
+
+    // 13. Assert: constructorInfo is not an abrupt completion.
+
+    // 14. Let F be constructorInfo.[[closure]]
+    F = constructorInfo.$Closure;
+
+    // 15. If ClassHeritage opt is present, set F’s [[ConstructorKind]] internal slot to "derived".
+    if (ast.superClass) {
+      F.$ConstructorKind = "derived";
+    }
+
+    // 16. Perform MakeConstructor(F, false, proto).
+    MakeConstructor(realm, F, false, proto);
+
+    // 17. Perform MakeClassConstructor(F).
+    MakeClassConstructor(realm, F);
+
+    // 18. Perform CreateMethodProperty(proto, "constructor", F).
+    CreateMethodProperty(realm, proto, "constructor", F);
+
+    let methods;
+    // 19. If ClassBody opt is not present, let methods be a new empty List.
+    if (ClassBody.length === 0) {
+      methods = [];
+    } else { // 20. Else, let methods be NonConstructorMethodDefinitions of ClassBody.
+      methods = NonConstructorMethodDefinitions(realm, ClassBody);
+    }
+
+    // 21. For each ClassElement m in order from methods
+    for (let m of methods) {
+      let status;
+      try {
+        // a. If IsStatic of m is false, then
+        if (!IsStatic(m)) {
+          // Let status be the result of performing PropertyDefinitionEvaluation for m with arguments proto and false.
+          PropertyDefinitionEvaluation(realm, m, proto, env, strictCode, false);
+        } else { // Else,
+          // Let status be the result of performing PropertyDefinitionEvaluation for m with arguments F and false.
+          PropertyDefinitionEvaluation(realm, m, F, env, strictCode, false);
+        }
+      } catch (e) {
+        if (e instanceof AbruptCompletion) {
+          status = e;
+        } else {
+          throw e;
+        }
+      }
+
+      // c. If status is an abrupt completion, then
+      if (status instanceof AbruptCompletion) {
+        // i. Set the running execution context's LexicalEnvironment to lex.
+        realm.getRunningContext().lexicalEnvironment = lex;
```

Comment:
The finally will take care of this, which means that you don't need the catch above and you can eliminate this if.